src/main.py

```diff
@@ -1,92 +1,81 @@
 from flask import Flask, render_template, abort
 from flask_talisman import Talisman
 from csp import csp
+import logging
 
 app = Flask(__name__)
 Talisman(app,
          content_security_policy=csp,
          content_security_policy_nonce_in=['script-src'])
 
-# TODO: Don't expect the same languages to be supported each year.
-SUPPORTED_LANGS = ('en', 'ja')
-SUPPORTED_YEARS = ('2019')
+DEFAULT_YEAR = '2019'
+DEFAULT_LANG = 'en'
+SUPPORTED_YEARS = {
+    '2019': ('en', 'ja')
+}
 
-##
-## English is implied when omitting a language code from the URL.
-## TODO: Should we require language codes to simplify this?
-##
 
 @app.route('/')
-def index():
-    return index_i18n('en')
-
-@app.route('/<year>/outline')
-def outline(year):
-    return outline_i18n('en', year)
-
-@app.route('/<year>/contributors')
-def contributors(year):
-    return contributors_i18n('en', year)
-
-@app.route('/<year>/methodology')
-def methodology(year):
-    return methodology_i18n('en', year)
-
-@app.route('/<year>/<chapter>/')
-def chapter(year, chapter):
-    return chapter_i18n('en', year, chapter)
-
-##
-## Internationalization (i18n)
-##
-
 @app.route('/<lang>/')
-def index_i18n(lang):
-    if lang not in SUPPORTED_LANGS:
-        abort(404)
+def index(lang=DEFAULT_LANG):
```

Comment:
I _really_ like this approach to combine the implicit and explicit langs. One hesitation I have is that it obscures the use case @MSakamaki raised in #52. It'd be nice to overwrite the default implicit lang from English to the user's Accept-Language preference, if supported.

I think the solution would be to initialize `lang=None` and do the Accept-Languages check when that's the case. Otherwise, default to DEFAULT_LANG.

Reply:
Got it! Should be easy enough to do, will give it a go in a few hours.

Reply:
Ok great. I'll go ahead and merge this as-is into the scaffolding branch. When you're ready you can make another PR for the Accept-Language change.